### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by chiphuyen**
Update README for:
- main NeMo
- NeMo core
- NeMo collections

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -14,63 +14,82 @@
 NVIDIA Neural Modules: NeMo
 ===========================
 
-Neural Modules (NeMo) is a framework-agnostic toolkit for building AI applications powered by Neural Modules. Current support is for PyTorch framework.
+NeMo (**Ne**ural **Mo**dules) is a toolkit for creating AI applications built around **neural modules**, conceptual blocks of neural networks that take *typed* inputs and produce *typed* outputs. Such modules typically represent data layers, encoders, decoders, language models, loss functions, or methods of combining activations.
```
Comment (okuchaiev) [inline README.rst]: The first phrase looks a strange when the actual README is rendered on GitHub (just navigate to "typo") branch to see. Maybe get rid of some bolding to fix it?

File: README.rst [hunk 2]
```diff
@@ -14,63 +14,82 @@
...
 
-A "Neural Module" is a block of code that computes a set of outputs from a set of inputs.
+NeMo makes it easy to combine and re-use these building blocks while providing a level of semantic correctness checking via its neural type system. As long as two modules have compatible inputs and outputs, it is legal to chain them together. An application built with NeMo is a **Directed Acyclic Graph** (DAG) of connected modules.
 
-Neural Modules’ inputs and outputs have Neural Type for semantic checking.
+NeMo's API is designed to be **framework-agnostic**, but currently only PyTorch is supported. We plan on supporting other frameworks in the near future.
```
Comment (okuchaiev) [inline README.rst]: we do plan eventually, but let's remove this phrase :)

File: README.rst [hunk 3]
```diff
@@ -14,63 +14,82 @@
...
-* `Collections` - NeMo comes with collections - related group of modules such as `nemo_asr` (for Speech Recognition) and `nemo_nlp` for NLP
+See `this video <https://nvidia.github.io/NeMo/>`_ for a quick walk-through.
 
 
 **Requirements**
 
 1) Python 3.6 or 3.7
 2) Pytorch 1.2 with GPU support
-3) NVIDIA APEX: https://github.com/NVIDIA/apex
+3) NVIDIA APEX. Install here: https://github.com/NVIDIA/apex
```
Comment (okuchaiev) [inline README.rst]: "Install from here:"

File: README.rst [hunk 4]
```diff
@@ -14,63 +14,82 @@
...
-        2. ``python setup.py install``
+``pip install .``
+
+1) Clone the repository ``git clone https://github.com/NVIDIA/NeMo.git``
+2) Go to NeMo folder and install the toolkit:
+
+.. code-block:: bash
+
+	cd NeMo
+	pip install .
```
Comment (okuchaiev) [inline README.rst]: should we change our "reinstall.sh" script too?

File: nemo/README.md [hunk 5]
```diff
@@ -7,17 +9,17 @@ NEMO is built around these principles:
...
-* :class:`NeuralType<nemo.core.neural_types.NeuralType>` captures expected (or actual) Tensor dimensions' semantics, dimensionality and more.
+* The input and output ports of each neural module and :class:`NmTensor<nemo.core.neural_types.NmTensor>` objects all have *Neural Type* - :class:`NeuralType<nemo.core.neural_types.NeuralType>`.
+* :class:`NeuralType<nemo.core.neural_types.NeuralType>` the semantics, axis order, and dimensions of a tensor and more.
 * Domain or module-specific *Callbacks* are used for evaluations and inspecting training performance.
 
 
 Built to run on GPUs
 --------------------
-On a toolkit-level NeMo provides:
+NeMo provides:
```
Comment (okuchaiev) [inline nemo/README.md]: "NeMo Core provides:"

File: README.rst (hunk 4 above)
Comment (chiphuyen) [reply README.rst]: yes!

Comment (okuchaiev) [reply README.rst]: in a separate PR then

**Pull request merged**